### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by mdiluz**
- README restructure and cleanup
- Source file rename and restructure
- Wine code full separation into wine file
- Include cleanup
- Comment cleanup

### CODE REVIEW

File: common/common-gpu.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ POSSIBILITY OF SUCH DAMAGE.
  */
 
 #pragma once
-#include "daemon_config.h"
+#define CONFIG_VALUE_MAX 256
```
Comment (aejsmith) [inline common/common-gpu.h]: Why the duplicated definition?

File: daemon/gamemode-context.c [hunk 2]
```diff
@@ -602,10 +602,12 @@ int game_mode_context_query_status(GameModeContext *self, pid_t client, pid_t re
  */
 static GameModeClient *game_mode_client_new(pid_t pid, char *executable)
 {
+	/* clang-format off */
```
Comment (aejsmith) [inline daemon/gamemode-context.c]: Why is this needed (and the other one)? These are in my preferred style already but is it not possible to get clang-format to do this?

**Comment by aejsmith** [comment]
Added a couple of comments, and can you rebase on master since there are now conflicts?

File: daemon/gamemode-context.c (hunk 2 above)
Comment (mdiluz) [reply daemon/gamemode-context.c]: I'm not 100% sure why but on multiple different installs, clang reformats these two sections differently to the Travis instance the CI is running. I've tried to coax it into doing it correctly with options but nothing seems to work.

File: common/common-gpu.h (hunk 1 above)
Comment (mdiluz) [reply common/common-gpu.h]: To split out the dependency between `common` and `daemon` code.

But yeah, it's a bit confusing now, I'll clean it up.

Comment (mdiluz) [reply common/common-gpu.h]: Fixed in d9b9141

**Comment by mdiluz** [comment]
Rebased and addressed comments
